### PR TITLE
Update Kotlin 1.2.0 -> 1.2.10

### PR DESCRIPTION
**Pull request by rock3r**
## Problem

We're using an outdated version of Kotlin, and we have a bunch of compile warnings.

## Solution

Update Kotlin to the latest stable version and address all the warnings/fix some formatting

### Paired with 

Nobody

### CODE REVIEW

File: .idea/inspectionProfiles/Squanchy.xml [hunk 1]
```diff
@@ -193,7 +193,7 @@
     <inspection_tool class="PointlessIndexOfComparison" enabled="true" level="WARNING" enabled_by_default="true" />
     <inspection_tool class="PointlessNullCheck" enabled="true" level="WARNING" enabled_by_default="true" />
     <inspection_tool class="PrivatePropertyName" enabled="true" level="WEAK WARNING" enabled_by_default="true">
-      <option name="namePattern" value="(?!m[A-Z])[a-z][A-Za-z\d]*" />
+      <option name="namePattern" value="(?!m[A-Z])_?[a-z][A-Za-z\d]*" />
```
Comment (rock3r) [inline .idea/inspectionProfiles/Squanchy.xml:196]: This allows for private backing fields for properties which conventionally begin with `_`, e.g., `_events`

File: app/src/main/java/net/squanchy/eventdetails/widget/EventDetailsLayout.kt [hunk 2]
```diff
@@ -15,14 +15,7 @@ import android.util.AttributeSet
...
 import android.widget.LinearLayout
-import kotlinx.android.synthetic.main.merge_event_details_layout.view.whenContainer
-import kotlinx.android.synthetic.main.merge_event_details_layout.view.whenTextView
-import kotlinx.android.synthetic.main.merge_event_details_layout.view.whereContainer
-import kotlinx.android.synthetic.main.merge_event_details_layout.view.whereTextView
-import kotlinx.android.synthetic.main.merge_event_details_layout.view.levelContainer
-import kotlinx.android.synthetic.main.merge_event_details_layout.view.levelTextView
-import kotlinx.android.synthetic.main.merge_event_details_layout.view.descriptionHeader
-import kotlinx.android.synthetic.main.merge_event_details_layout.view.descriptionTextView
+import kotlinx.android.synthetic.main.merge_event_details_layout.view.*
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/eventdetails/widget/EventDetailsLayout.kt:18]: IDEA default settings, I don't see why not tbh /shrug

File: app/src/main/java/net/squanchy/schedule/SchedulePageView.kt [hunk 3]
```diff
@@ -100,18 +103,14 @@ class SchedulePageView : CoordinatorLayout, Loadable {
...
 
-            val tabCount = tabLayout.tabCount
-            for (i in 0 until tabCount) {
-                val tab = tabLayout.getTabAt(i)
-                if (tab == null || hasTypefaceSpan(tab.text)) {
-                    continue
-                }
-                tab.text = tab.text?.applyTypeface(typeface)
-            }
+            (0 until tabLayout.tabCount)
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/schedule/SchedulePageView.kt:110]: The IDE really wanted me to do this /shrug

File: app/src/main/java/net/squanchy/venue/VenueInfoPageView.kt [hunk 4]
```diff
@@ -23,28 +20,29 @@ import net.squanchy.navigation.Navigator
...
+), Loadable {
 
     private lateinit var navigator: Navigator
     private lateinit var service: VenueInfoService
     private lateinit var imageLoader: ImageLoader
 
     private var subscription: Disposable? = null
 
+    init {
+        if (!isInEditMode) {
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/venue/VenueInfoPageView.kt:36]: One cannot have `return`s in `init` blocks

File: app/src/main/java/net/squanchy/speaker/widget/SpeakerDetailsLayout.kt [hunk 5]
```diff
@@ -28,13 +25,13 @@ class SpeakerDetailsLayout(context: Context, attrs: AttributeSet) : LinearLayout
         speakerBio.text = parseHtml(speaker.bio)
     }
 
-    @TargetApi(Build.VERSION_CODES.N)
-    // The older fromHtml() is only called pre-24
+    @TargetApi(Build.VERSION_CODES.N)     // The older fromHtml() is only called pre-24
```
Comment (fourlastor) [inline app/src/main/java/net/squanchy/speaker/widget/SpeakerDetailsLayout.kt:28]: this has now logic in a bunch of places, worth extracting a parseHtml top level function?

**Review by fourlastor — APPROVED** [review]
Only one comment

File: app/src/main/java/net/squanchy/schedule/SchedulePageView.kt [hunk 6]
```diff
@@ -24,25 +24,28 @@ import net.squanchy.support.unwrapToActivityContext
 import timber.log.Timber
 
 
-class SchedulePageView : CoordinatorLayout, Loadable {
+@Suppress("UNUSED_ANONYMOUS_PARAMETER")
+class SchedulePageView @JvmOverloads constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int = 0) : CoordinatorLayout(
```
Comment (quiro91) [inline app/src/main/java/net/squanchy/schedule/SchedulePageView.kt]: I think it should be 
```
class SchedulePageView @JvmOverloads constructor(
    context: Context, 
    attrs: AttributeSet, 
    defStyleAttr: Int = 0
) : CoordinatorLayout(context, attrs, defStyleAttr), Loadable
```

File: app/src/main/java/net/squanchy/venue/VenueInfoPageView.kt [hunk 7]
```diff
@@ -23,28 +20,29 @@ import net.squanchy.navigation.Navigator
...
-        }
-
-        val activity = unwrapToActivityContext(getContext())
-        val component = venueInfoComponent(activity)
-
-        navigator = component.navigator()
-        service = component.service()
-        imageLoader = imageLoaderComponent(activity).imageLoader()
-    }
+class VenueInfoPageView @JvmOverloads constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int = 0) : CoordinatorLayout(
```
Comment (quiro91) [inline app/src/main/java/net/squanchy/venue/VenueInfoPageView.kt]: Same here

File: app/src/main/java/net/squanchy/speaker/widget/SpeakerDetailsLayout.kt (hunk 5 above)
Comment (quiro91) [reply app/src/main/java/net/squanchy/speaker/widget/SpeakerDetailsLayout.kt:28]: Agree, but probably the end goal is to use Dante everywhere

**Review by quiro91 — CHANGES_REQUESTED** [review]
A couple of styling changes and we are good to go

File: app/src/main/java/net/squanchy/speaker/widget/SpeakerDetailsLayout.kt (hunk 5 above)
Comment (rock3r) [reply app/src/main/java/net/squanchy/speaker/widget/SpeakerDetailsLayout.kt:28]: What Lorenzo says, aiming to replace them all. Not worth doing it now (it's 4 copies right now btw)

File: app/src/main/java/net/squanchy/schedule/SchedulePageView.kt (hunk 6 above)
Comment (rock3r) [reply app/src/main/java/net/squanchy/schedule/SchedulePageView.kt]: Yeah, autoformatter screwed me :(

**Review by quiro91 — APPROVED** [review]

**Pull request merged**